osquery/config/config.cpp

```diff
@@ -530,6 +541,54 @@ void stripConfigComments(std::string& json) {
   json = sink;
 }
 
+std::map<std::string, std::string> Config::restoreConfigBackup() {
+  if (!FLAGS_config_enable_backup) {
+    return std::map<std::string, std::string>();
+  }
+
+  VLOG(1) << "Restoring backed up config from the database";
+  std::vector<std::string> keys;
+  std::map<std::string, std::string> config;
+
+  WriteLock lock(config_backup_mutex_);
+  scanDatabaseKeys(kPersistentSettings, keys, kConfigPersistencePrefix);
+
+  for (const auto& key : keys) {
+    std::string value;
+    Status status = getDatabaseValue(kPersistentSettings, key, value);
+    if (!status.ok()) {
+      continue;
```

Comment:
Should backup function fail on this error? Or write down an error log message at least.

Reply:
Nice.
I don't think there can be anything actionable done in this case. Restoring at least things that are not corrupted looks the best case scenario for me. Also, all backup(including corrupted ones) will be removed when config is updated, so this problem will not persist.
Will add VLOG

Reply:
done